sound/soc/sof/sof-priv.h

```diff
@@ -120,10 +120,10 @@ struct snd_sof_dsp_ops {
 
 	/* memcpy IO */
 	void (*block_read)(struct snd_sof_dev *sof_dev, u32 bar,
-			   u32 offset, void *dest,
+			   size_t offset, void *dest,
 			   size_t size); /* mandatory */
 	void (*block_write)(struct snd_sof_dev *sof_dev, u32 bar,
-			    u32 offset, void *src,
+			    size_t offset, void *src,
```

Comment:
I don't think this is right. When doing rea/dwrites from registers, we use integer offsets. I don't see the point of using size_t here.

Reply:
@plbossart this is indeed a bit less clear than "trivial" `size_t` use-cases. I even seem to remember a special `offset_t` type, but maybe that wasn't in the kernel or not in this universe. However, there are definitely cases of `size_t offset` in the kernel, even in `sound/`, even in SOF (without this patch). Just `grep "size_t offset" sound/`. And I actually see logic behind such uses too: both size and offset are "numbers of bytes." If `offset` is in some abstract elements (words, longs, objects), then `size_t` wouldn't be a good fit. However, where bytes are counted I do think `(s)size_t` fit well enough. And here it is a number of bytes, isn't it?

Reply:
I agree that when counting bytes size_t is fine.
But to represent a register address it's confusing